app/models/rubygem.rb

```diff
@@ -137,7 +139,12 @@ def indexed_versions?
 
   def owned_by?(user)
     return false unless user
-    ownerships.exists?(user_id: user.id)
+    ownerships.where(user_id: user.id).exists?
+  end
+
+  def unconfirmed_ownership?(user)
+    return false unless user
+    ownerships_including_unconfirmed.unconfirmed.where(user_id: user.id).exists?
```

Comment:
I think we can drop line 146 and just use `owners_including_unconfirmed.include? user`.  Probably change function name to `unconfirmed_owner?`. please do some research on this and see if there is any issue with using `include?`.

Same for `owned_by?`

Reply:
Umm it can be done for `owned_by?` but not for `unconfirmed_ownership?` because `owners_including_unconfirmed` includes owners which are confirmed as well. 
We can do (owners_including_unconfirmed - owners) or add a new association to make it work though. What do you suggest?

Reply:
> add a new association to make it work though

Probably not as useful just for single usecase. 
Move `unconfirmed.where(user_id: user.id).exists?` logic to ownership model tho. Rubygem doesn't need to know about all these details.